### PR TITLE
telink: tl321x: adopt and update drivers

### PR DESCRIPTION
**Pull request by damien0x0023**
- update gpio driver to enable interrupts.
- adopt i2c driver to enable connection with periphery sensor.
- adopt adc driver to enable voltage measurement.

### CODE REVIEW

File: boards/riscv/tl321x/tl3218x-common.dtsi [hunk 1]
```diff
@@ -153,7 +153,7 @@
 };
 
 &gpiob {
-	interrupts = <37 1>;
+	interrupts = <47 1>;
```
Comment (pulin1103) [inline boards/riscv/tl321x/tl3218x-common.dtsi:156]: Why use IRQ_GPIO_IRQ1 for gpiob, Can we use IRQ_GPIO_IRQ0 for this port?

Comment (charlie-ai) [reply boards/riscv/tl321x/tl3218x-common.dtsi:156]: Randomly assigned, with no restrictions

Comment (damien0x0023) [reply boards/riscv/tl321x/tl3218x-common.dtsi:156]: Yes, we can assign IRQ_GPIO_IRQ0 for gpiob.
Leave IRQ_GPIO_IRQ1 to other ports like port a if we'd like to.

**Review by andy-liu-telink — APPROVED** [review]
Approve

**Pull request merged**